src/shared/utils/url.ts

```diff
@@ -10,21 +9,46 @@ export const getLocationForPathname = (pathname: string) => {
   return found?.[0];
 };
 
-/**
-Wrapper around query-string's parse function which takes the output of this and
-returns only the first value of a parameter if it is an array.
+type QueryStringParamsRecord = Record<
+  string,
+  string | string[] | number | number[] | boolean | undefined | null
+>;
+export type QueryStringArg = string | QueryStringParamsRecord | URLSearchParams;
 
-@param query - The query string to parse.
-@param options - Options to pass query-string's parse 
-*/
-export const parseQueryString = (
-  query: string,
-  options?: queryString.ParseOptions
-) => {
-  const parsed = queryString.parse(query, options);
-  const parsedWithoutArrayValues: { [key: string]: string | null } = {};
-  for (const [key, value] of Object.entries(parsed)) {
-    parsedWithoutArrayValues[key] = Array.isArray(value) ? value[0] : value;
+export const stringifyQuery = (...args: QueryStringArg[]) => {
+  // This returns a query string by iterating over the args and creating a combined
+  // URLSearchParams instance. If a parameter key already exists it will be overwritten
+  // by the new value. If the new value is undefined or null then the parameter will
+  // be removed. Arrays will be returned as comma separated strings.
+  const combined = new URLSearchParams();
+  for (const arg of args) {
+    const iter =
+      // Query string
+      (typeof arg === 'string' && new URLSearchParams(arg)) ||
```

Comment:
Not sure what `&& new URLSearchParams(arg)` is for, it will always return a truthy value as it will return an URLSearchParameter object, even if there is nothing in `arg`.